protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/MailCmdHandler.java

```diff
@@ -233,34 +232,40 @@ private Response doMAILFilter(SMTPSession session, String argument) {
                 }
 
                 try {
-                    senderAddress = new MailAddress(sender);
+                    senderAddress = MaybeSender.of(new MailAddress(sender));
                 } catch (Exception pe) {
                     LOGGER.info("Error parsing sender address: {}", sender, pe);
                     return SYNTAX_ERROR_ADDRESS;
                 }
             }
-            if ((senderAddress == null) || 
-                    ((senderAddress.getLocalPart().length() == 0) && (senderAddress.getDomain().name().length() == 0))) {
-                senderAddress = MailAddress.nullSender();
+            if (isNullSender(senderAddress)) {
+                senderAddress = MaybeSender.nullSender();
```

Comment:
this is ... non-sense

Reply:
I rewrote this crazy variable re-affectation stuff.